src/GeniusYield/Test/Privnet/Ctx.hs

```diff
@@ -239,17 +219,17 @@ addRefScriptCtx :: Ctx                 -- ^ Given context.
                 -> User                -- ^ User which will execute the transaction (if required).
                 -> GYScript 'PlutusV2  -- ^ Given script.
                 -> IO GYTxOutRef       -- ^ Returns the reference for the desired output.
-addRefScriptCtx ctx user script = do
-  txBodyRefScript <- ctxRunF ctx user $ Limbo.addRefScript script
+addRefScriptCtx ctx user script = ctxRun ctx user $ do
+  txBodyRefScript <- Limbo.addRefScript script >>= traverse buildTxBody
   case txBodyRefScript of
-    Left ref -> return ref
+    Left ref -> pure ref
     Right body -> do
       let refs = Limbo.findRefScriptsInBody body
       ref <- case Map.lookup (Some script) refs of
         Just ref -> return ref
-        Nothing  -> fail "Shouldn't happen: no ref in body"
-      void $ submitTx ctx user body
-      return ref
+        Nothing  -> throwAppError $ someBackendError "Shouldn't happen: no ref in body"
+      signAndSubmitConfirmed_ body
```

Comment:
`GYTxMonadIO` (should really be used for bot-like usecase, use `GYTxBuilderMonadIO` for app-like usecases) now contains the user keys. So `signAndSubmitConfirmed_` is a helper that uses the payment key.